### DISCUSSION TITLE
How to wait for best scored image before sending to MQTT

### DISCUSSION THREAD

**Discussion by esvaf** [blakeblackshear/frigate#10362] [General]
Hi there!
Is it possible to send only the best frames within N seconds of an event to the MQTT? The same way it is done to display events. At the moment I am receiving snapshots for alerting system from MQTT and there are often bad photos from the start of event or even duplicates for one event, while in the frigate interface I see good snapshots for all events. I think i need option like "wait N seconds for best score image before send to MQTT" but i can't fnd out how to configure it.

**Comment by NickM-27** [comment]
Are you referring to the images that are posted to mqtt or are you using mqtt /events topic and pulling the image yourself

**Comment by esvaf** [comment]
> Are you referring to the images that are posted to mqtt or are you using mqtt /events topic and pulling the image yourself

Now I use images posted to topics like frigate/cameraname/car/snapshot

**Reply by NickM-27** [reply]
you can set required zones for mqtt and it will only send those when an object is detected in the zones

**Reply by esvaf** [reply]
It can be a solution but not best because in some cases i will never receive notification. I want to receive all events of cars in my example, but wait for the best image (not more than 15 sec for example). It's work really fine for snapshot in event preview for UI - like it. I don't want to loose information if car went only side of screen, but if it went to the center I want to receive image with best angle.

**Reply by NickM-27** [reply]
The snapshots that are chosen for the event UI and the images sent via mqtt are identical, they are saved for both places at the same time. Odds are you just aren't seeing the number of times that it changes the image

**Reply by esvaf** [reply]
You mean that I need to update image on the side of my software? How i can define that it's a update for current event, not new one?

**Reply by esvaf** [reply]
Ok, i will try to get info from event message and get image from http api.... I think it's only one possible solution
